services/libs/aster-std/src/syscall/chown.rs

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::fs::{
+    file_table::FileDescripter,
+    fs_resolver::{FsPath, AT_FDCWD},
+    utils::PATH_MAX,
+};
+use crate::log_syscall_entry;
+use crate::prelude::*;
+use crate::process::{Gid, Uid};
+use crate::util::read_cstring_from_user;
+
+use super::SyscallReturn;
+use super::{SYS_FCHOWN, SYS_FCHOWNAT};
+
+pub fn sys_fchown(fd: FileDescripter, uid: u32, gid: u32) -> Result<SyscallReturn> {
```

Comment:
> If the owner or group is specified as -1, then that ID is not
       changed.

By using `u32` as the types, the implementation falls short of handling the corner case of `-1`.

Reply:
Thank you for this reminder; I didn't even notice that instruction.